mogp_gpu/src/kernel.cu

```diff
@@ -0,0 +1,408 @@
+#include "kernel.hpp"
+#include <stdio.h>
+#include <iostream>
+
+/// Device functions, and global functions (also, confusingly called "kernels") preceded by
+/// "sqexp" for the SquaredExponential kernel, and "mat" for the Matern52 kernel.
```

Comment:
We may also at some point add the Matern 3/2 kernel, so may be worth using "mat52" in all of these prefixes?

Reply:
Good idea - done! :)